service/src/main/java/org/databiosphere/workspacedataservice/controller/CloningController.java

```diff
@@ -39,4 +40,11 @@ public ResponseEntity<Job<BackupResponse>> getBackupStatus(@PathVariable("versio
         var response = backupRestoreService.checkBackupStatus(trackingId);
         return new ResponseEntity<>(response, HttpStatus.OK);
     }
+
+    @GetMapping("/clone/{version}")
+    public ResponseEntity<Job<CloneResponse>> getCloningStatus(@PathVariable("version") String version) {
+        validateVersion(version);
+        var response = backupRestoreService.checkCloneStatus();
+        return new ResponseEntity<>(response, HttpStatus.OK);
```

Comment:
how about this?
```suggestion
        var status = (response == null) ? HttpStatus.NOT_FOUND : HttpStatus.OK;
        return new ResponseEntity<>(response, status);
```
this makes it a little more REST-ful and hopefully more friendly for callers

Reply:
ah good call! that would be better!